apiClient.go

```diff
@@ -32,51 +34,93 @@ type apiClient struct {
+func (c *apiClient) OrganizationProjects(ctx context.Context, orgID string) (Projects, error) {
 
-	getStatusURL := fmt.Sprintf("%v/...", c.apiBaseURL)
+	var wrapper struct {
+		Projects `json:"projects"`
+	}
+
+	organizationProjectsURL := fmt.Sprintf("%v/org/%s/projects", c.apiBaseURL, orgID)
 
 	headers := map[string]string{
-		"Content-Type": "application/json",
+		"Authorization": fmt.Sprintf("token %s", c.apiToken),
+		"Content-Type":  "application/json",
 	}
 
-	responseBody, err := c.getRequest(getStatusURL, nil, headers)
+	response, err := c.getRequest(organizationProjectsURL, nil, headers, 200)
 	if err != nil {
-		log.Error().Err(err).Str("url", getStatusURL).Msgf("Failed retrieving snyk status")
-		return
+		log.Fatal().Msgf("Failed retrieving projects from snyk")
 	}
 
-	var statusResponse struct {
-		Status string `json:"status"`
+	defer response.Body.Close()
+
+	bodyBytes, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		log.Info().Msgf(err.Error())
+		log.Fatal().Msgf("Error reading body of snyk response")
 	}
 
 	// unmarshal json body
-	err = json.Unmarshal(responseBody, &statusResponse)
+	err = json.Unmarshal(bodyBytes, &wrapper)
 	if err != nil {
-		log.Error().Err(err).Str("body", string(responseBody)).Str("url", getStatusURL).Msgf("Failed unmarshalling snyk status response")
-		return
+		log.Info().Msgf(err.Error())
+		log.Fatal().Msgf("Failed unmarshalling snyk response body")
 	}
 
-	return statusResponse.Status, nil
-}
+	return wrapper.Projects, nil
 
-func (c *apiClient) getRequest(uri string, requestBody io.Reader, headers map[string]string, allowedStatusCodes ...int) (responseBody []byte, err error) {
-	return c.makeRequest("GET", uri, requestBody, headers, allowedStatusCodes...)
 }
 
-func (c *apiClient) postRequest(uri string, requestBody io.Reader, headers map[string]string, allowedStatusCodes ...int) (responseBody []byte, err error) {
-	return c.makeRequest("POST", uri, requestBody, headers, allowedStatusCodes...)
+func (c *apiClient) ProjectVulnerabilities(ctx context.Context, orgID, projectID string, filters *projectFilters) (Vulnerabilities, error) {
+	pIssues := &projectIssuesWrapper{}
+
+	if filters == nil {
+		filters = defaultFilters()
+	}
+
+	data := struct {
+		Filters *projectFilters `json:"filters"`
+	}{
+		filters,
+	}
+
+	jsonBytes, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+	body := bytes.NewReader(jsonBytes)
+
+	projectVulnerabilitiesURL := fmt.Sprintf("%v/org/%s/project/%s/issues", c.apiBaseURL, orgID, projectID)
+
+	headers := map[string]string{
+		"Authorization": fmt.Sprintf("token %s", c.apiToken),
+		"Content-Type":  "application/json",
+	}
+
+	response, err := c.postRequest(projectVulnerabilitiesURL, body, headers, 200)
+	if err != nil {
+		log.Fatal().Msgf("Failed retrieving vulnerabilities from snyk")
+	}
+
+	defer response.Body.Close()
+
+	err = json.NewDecoder(response.Body).Decode(pIssues)
```

Comment:
What's the reason to use `Decode` here and `Unmarshal` elsewhere?

Reply:
No particular reason. Well spotted. I was looking at different implementations and ended up mixing different approaches. I will change it to `Unmarshal`.

Reply:
In that case you can move the reading of the response body back into the shared method for making the actual request.